llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -4011,15 +4011,16 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
       // Fill in any known bits from range information. There are 3 types being
       // used. The results VT (same vector elt size as BitWidth), the loaded
       // MemoryVT (which may or may not be vector) and the range VTs original
-      // type. The range matadata needs the full range (i.e
+      // type. The range metadata needs the full range (i.e
       // MemoryVT().getSizeInBits()), which is truncated to the correct elt size
       // if it is know. These are then extended to the original VT sizes below.
       if (const MDNode *MD = LD->getRanges()) {
+        ConstantInt *Lower = mdconst::extract<ConstantInt>(MD->getOperand(0));
+
+        Known0 = Known0.trunc(Lower->getBitWidth());
```

Comment:
Use a different temporary KnownBits with the correct size instead of truncation Known0? computeKnownBitsFromRangeMetadata can't do anything useful with the incoming value

Reply:
Use local KnownBits with correct size instead of truncation.